src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -674,7 +674,7 @@ void checkConstraintLanguageVersionAndRuntimeSupportForConversion(SyntaxNode syn
             }
             else
             {
-                if ((collectionTypeKind is CollectionExpressionTypeKind.ArrayInterface or CollectionExpressionTypeKind.List) ||
+                if ((collectionTypeKind is CollectionExpressionTypeKind.ArrayInterface) ||
```

Comment:
Do we need to check presence of these members when `collectionTypeKind` is `CollectionInitializer`?

Reply:
We should only be using these methods for collection initializers when the target is `List<T>`, and in those cases, we'll bind to the methods in initial binding.